Adapters/Vungle/ISVungleAdapter/ISVungleAdapter.m

```diff
@@ -205,30 +163,27 @@ - (void)onNetworkInitCallbackSuccess {
     LogAdapterDelegate_Internal(@"");
     
     // rewarded video
-    NSArray *rewardedVideoPlacementIDs = _rewardedVideoPlacementIdToSmashDelegate.allKeys;
-    
+    NSArray *rewardedVideoPlacementIDs = _rewardedVideoPlacementIdToSmashRouter.allKeys;
     for (NSString *placementId in rewardedVideoPlacementIDs) {
         if ([_rewardedVideoPlacementIdsForInitCallbacks hasObject:placementId]) {
-            [[_rewardedVideoPlacementIdToSmashDelegate objectForKey:placementId] adapterRewardedVideoInitSuccess];
+            [[_rewardedVideoPlacementIdToSmashRouter objectForKey:placementId] rewardedVideoInitSuccess];
         } else {
             [self loadRewardedVideoInternalWithPlacement:placementId];
```

Comment:
@YueVungle Should this be removed? How does the adapter know that the placement id is for rewarded?

Reply:
@stanley-vungle Hi Stanley, I think we might need this, if the mediation needs to get a callback for init Rewarded Video, we need `_rewardedVideoPlacementIdsForInitCallbacks` set to find that delegate and notify them. If the mediation doesn't need to get a callback for init Rewarded Video, we just load RewardedVideo or notify the delegate RewardedVideo availability is changed to NO. Please check below two methods:

```
// used for flows when the mediation needs to get a callback for init
- (void)initRewardedVideoForCallbacksWithUserId:(NSString *)userId
                                  adapterConfig:(ISAdapterConfig *)adapterConfig
                                       delegate:(id<ISRewardedVideoAdapterDelegate>)delegate;
```

```
// used for flows when the mediation doesn't need to get a callback for init
- (void)initAndLoadRewardedVideoWithUserId:(NSString *)userId
                             adapterConfig:(ISAdapterConfig *)adapterConfig
                                  delegate:(id<ISRewardedVideoAdapterDelegate>)delegate
```
Thanks a lot. :)